spec.bs

```diff
@@ -1171,34 +1169,81 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
     1. If [=recursively wait until configuration input promises resolve=] given |auctionConfig| returns failure, return failure.
     1. If |compWinner| is not null:
       1. Run [=score and rank a bid=] with |auctionConfig|, |compWinner|, |leadingBidInfo|,
-        |decisionLogicScript|, null, "top-level-auction", null, and |settings|'s [=environment/top-level origin=].
-
+        |decisionLogicScript|, null, null, "top-level-auction", null, and |settings|'s
+        [=environment/top-level origin=].
     1. Decrement |pendingComponentAuctions| by 1.
   1. Wait until |pendingComponentAuctions| is 0.
   1. If |leadingBidInfo|'s [=leading bid info/leading bid=] is null, return null.
   1. Let |winningComponentConfig| be |leadingBidInfo|'s [=leading bid info/auction config=].
   1. Set |leadingBidInfo|'s [=leading bid info/auction config=] to |auctionConfig|.
   1. Set |leadingBidInfo|'s [=leading bid info/component seller=] to |winningComponentConfig|'s
     [=auction config/seller=].
+  1. Let |topLevelDirectFromSellerSignals| be the result of running [=get direct from seller signals=]
+    with |auctionConfig|'s [=auction config/expects direct from seller signals header ad slot=],
+    |auctionConfig|'s [=auction config/seller=], |auctionConfig|'s
+    [=auction config/direct from seller signals header ad slot=], and |capturedAuctionHeaders|.
+  1. Let |topLevelSellerDirectFromSellerSignals| be a {{DirectFromSellerSignalsForSeller}} with the

```

Comment:
I wonder if it would be more clear to pull this into another algorithm, rather than repeat it a few times?

Looks like you already have a similar helper for buyers?